packages/cohort_banking/src/callbacks/oo_installer_single_query.rs

```diff
@@ -0,0 +1,186 @@
+            WHERE ba."number" IN (($1)::TEXT, ($2)::TEXT)
+                AND EXISTS (SELECT 1 FROM cohort_snapshot cs WHERE cs."version" >= ($5)::BIGINT)
+                AND ba."version" < ($4)::BIGINT
+            RETURNING
+                ba."number", ba."version" as "new_version", (null)::BIGINT as "version", (SELECT cs."version" FROM cohort_snapshot cs) as "snapshot"
+        )
+        SELECT * FROM bank_accounts_temp
+        UNION
+        SELECT
+            ba."number", (null)::BIGINT as "new_version", ba."version" as "version", cs."version" as "snapshot"
+        FROM
+            bank_accounts ba, cohort_snapshot cs
+        WHERE ba."number" IN (($1)::TEXT, ($2)::TEXT)
+            "#;
+
+        let params: &[&(dyn ToSql + Sync)] = &[
+            &self.request.from,
+            &self.request.to,
+            &self.request.amount,
+            &(new_version as i64),
+            &(safepoint as i64),
+        ];
+
+        let result = self
+            .database
+            .query_many(sql, params, |row| {
+                let nr = row
+                    .try_get::<&str, String>("number")
+                    .map_err(|e| DatabaseError::deserialise_payload(e.to_string(), "Cannot read account name".into()))?;
+                let new_ver = row
+                    .try_get::<&str, Option<i64>>("new_version")
+                    .map_err(|e| DatabaseError::deserialise_payload(e.to_string(), "Cannot read new_version column".into()))?;
+                let version = row
+                    .try_get::<&str, Option<i64>>("version")
+                    .map_err(|e| DatabaseError::deserialise_payload(e.to_string(), "Cannot read version column".into()))?;
+                let snapshot = row
+                    .try_get::<&str, i64>("snapshot")
+                    .map_err(|e| DatabaseError::deserialise_payload(e.to_string(), "Cannot read snapshot column".into()))?;
+
+                Ok((nr, new_ver, version, snapshot))
+            })
+            .await
+            .map_err(|e| e.to_string())?;
+
+        if result.is_empty() {
+            // there were no items found to work with
+            log::warn!(
+                "No bank accounts where found by these IDs: {:?}",
+                (self.request.from.clone(), self.request.to.clone())
+            );
+            let c = Arc::clone(&self.counter_oo_no_data_found);
+            let _ = tokio::spawn(async move {
+                c.add(&Context::current(), 1, &[]);
+            })
+            .await;
+            return Ok(OutOfOrderInstallOutcome::InstalledAlready);
+        }
+
+        // Quickly grab the snapshot to check whether safepoint condition is satisfied. Any row can be used for that.
+        let (_, _, _, snapshot) = &result[0];
+        if (*snapshot as u64) < safepoint {
+            return Ok(OutOfOrderInstallOutcome::SafepointCondition);
+        }
+
+        let c = Arc::clone(&self.counter_oo_installs);
+        let _ = tokio::spawn(async move {
```

Comment:
Why `await` this? 

Just run it without awaiting, as you don't really need worry about what is returned. 
This way rest of the code can continue executing without waiting for this handler to complete:-

```rust
tokio::spawn(async move {
            c.add(&Context::current(), 1, &[]);
        })
```

Reply:
You are right, I am changing it now.